src/js/tabs/motors.js

```diff
@@ -407,6 +407,8 @@ motors.initialize = async function (callback) {
                         return;
                     }
                 }
+                // if every item in the motor array (current size limit: 8) is populated, we have that many motors
+                self.numberOfValidOutputs = FC.MOTOR_DATA.length;
```

Comment:
This can be removed if we initialize line 394 with firmware value instead of model value as in

`self.numberOfValidOutputs = FC.MOTOR_CONFIG.motor_count;`

Also line 393 should then be `// initialize with firmware supplied motor_count`.

Validation routine then still works.